.travis.yml

```diff
@@ -1,33 +1,78 @@
-sudo: required
+# File managed by pluginsync
+sudo: true
 language: go
 go:
-- 1.5.4
-- 1.6.2
+- 1.6.3
+- 1.7.1
+addons:
+  apt:
+    packages:
+    - debhelper
+    - cmake
+    - libgeoip-dev
+    - protobuf-compiler
 services:
-- docker
-before_install:
-- go get github.com/tools/godep
-- if [ ! -d $SNAP_PLUGIN_SOURCE ]; then mkdir -p $HOME/gopath/src/github.com/intelsdi-x; ln -s $TRAVIS_BUILD_DIR $SNAP_PLUGIN_SOURCE; fi # CI for forks not from intelsdi-x
+  - docker
 env:
   global:
-    - SNAP_PLUGIN_SOURCE=/home/travis/gopath/src/github.com/intelsdi-x/snap-plugin-publisher-heka
+  - ORG_PATH=/home/travis/gopath/src/github.com/intelsdi-x
+  - SNAP_PLUGIN_SOURCE=/home/travis/gopath/src/github.com/${TRAVIS_REPO_SLUG}
   matrix:
-    - TEST=unit
-    - TEST=integration
-addons:
-  apt:
-    packages:
-      - debhelper # required for trusty VM
-      - cmake
-      - libgeoip-dev
-      - protobuf-compiler
+  - TEST_TYPE=small
+  - TEST_TYPE=build
+matrix:
+  exclude:
+  - go: 1.6.3
+    env: TEST_TYPE=build
+before_install:
+- "[[ -d $SNAP_PLUGIN_SOURCE ]] || mkdir -p $ORG_PATH && ln -s $TRAVIS_BUILD_DIR $SNAP_PLUGIN_SOURCE"
 install:
-- export TMPDIR=$HOME/tmp
-- mkdir -p $TMPDIR
-- cd $SNAP_PLUGIN_SOURCE # change dir into source
-- make
+- cd $SNAP_PLUGIN_SOURCE
+- make deps
```

Comment:
If we decide that old types should be added to test matrix in .travis.yml then this line should be changed to `make`, for now heka tests require building of plugin. Should I open an issue about it?